src/compiler/checker.ts

```diff
@@ -46483,6 +46483,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return undefined;
     }
 
+    function shouldParenthesizeConstantValue(node: AccessExpression) {

```

Comment:
Do we need a new function here? I wonder if this could be implemented in `parenthesizeLeftSideOfAccess`